### DISCUSSION TITLE
Too many bugs!!!

### DISCUSSION THREAD

**Discussion by vincode-io** [vincode-io/Zavala#230] [General]
Does it seem like Zavala is too buggy? Well... that's because it is. I hate saying that. I take pride in my work and Zavala simply doesn't live up to my standards.

I try to keep Zavala as simple to use as possible. That doesn't mean that coding something like Zavala is simple. It's very complex. Zavala runs on 3 different platforms, iPhones, iPads, and Macs. Although I'm able to share most of the code between all 3, each is slightly different in how they work. iCloud syncing for anything is hard. For an outline, it is crazy hard.

I need help! Please, if you care about the quality of Zavala, consider helping with testing. Sign up for [TestFlight](https://testflight.apple.com/join/iFbND23m) and provide feedback.

You can also email me using the Provide Feedback function in Zavala. I can't fix the bug I don't know about. This, TestFlight feedbacks, and [Github Issues](https://github.com/vincode-io/Zavala/issues) is how you can tell me about them.

Let's make Zavala the best outliner out there.

**Comment by jgrisham** [comment]
Thank you for putting detailed release notes into TestFlight!

_(so few developers bother entering anything there anymore, let alone consider adding useful levels of detail that changes with each build)_

**Reply by vincode-io** [reply]
It's not hard to do. As I add features and fix bugs, I add them to a release outline in Zavala. Then when doing a release, I can just drag the outline into the correct field in a web form and I'm done! It's one of the ways that I dogfood Zavala.

**Comment by saasa** [comment]
To be honest, the only real gripe I have with Zavala is the somewhat finicky controls on iOS. 

- The indent buttons, swiping the lines, these are too tiny to reliably operate on the move even on a big screen of 15 Pro Max.
- The hold-and-move reordering is also somewhat unreliable, especially what detecting an item to become a child of. 
- I recently realized the link items sometimes(?) cannot be moved by drawing at all. 

Anyway, I installed the TestFlight version and will try to be of assistance.

**Comment by vincode-io** [comment]
@saasa Thanks for the feedback!
